### DISCUSSION TITLE
call failed with status BadNotImplemented

### DISCUSSION THREAD

**Discussion by francy328** [eclipse-milo/milo#1276] [General]
I've tried to add namespaces to the example server, like as been done in https://github.com/kevinherron/machine-example/tree/master, (just changed some xml file).
I start the server and when I connect with UaExpert I see what I'm expecting to view, and that is really fantastic:
![image](https://github.com/eclipse/milo/assets/23223131/33ebf6d3-2d12-4c0d-9a1f-bc454abd11a7)

But when I call I method like GetLatestResult i get:

![image](https://github.com/eclipse/milo/assets/23223131/dd49923c-d42c-43cf-9a71-24975e9c7d88)

On the server side no log at all, So my question is Where to start in order to implement that method? Should I define a Listener on some node? and if that is true where I can find an example?

**Comment by kevinherron** [comment]
You need to get that Node instance somehow and install an InvocationHandler on it.

Alternatively, you could override the `call` implementation of your AddressSpace, and try to handle the CallMethodRequests yourself, figuring out which object/method is which based on the ids in the requests.

Or if it's a `ManagedAddressSpace` subclass, override `getImplementationHandler`.

**Reply by francy328** [reply]
Thx Kevin for replay;

I'm trying to figure out as you indicated but no success so far!


**for the First indication**


I know I have my node in the server, but how I can find it? 

I've tried:
```
       NodeId objectId2 = NodeId.parse("ns=2;s=JoiningProcessManagement");
   	server.getAddressSpaceManager().getManagedNode(objectId2);
```
but the last line dosn't give me an object like i have in:
```
 UaObjectNode myObject = (UaObjectNode) getNodeFactory().createNode(
                newNodeId("HelloWorld/MyObject"),
                objectTypeNode.getNodeId()
            );
```
in order to associate an hander to it like this:
```
List<UaMethodNode> listMethod = myObject.getMethodNodes();
            listMethod.get(0).setInvocationHandler(sqrtMethod);
```

in the last snippet myObject is of type UaObjectNode.

**for the second indication:**
how I can get addressSpace ? on the server side there is server.getAddressSpaceManager() but I think is not the same



**for the third indication:**
Yes the class ModeledAddressSpace is a subclass of ManagedAddressSpace 
but there is not method called getImplementationHandler, may be you say getInvocationHandler?

**Reply by kevinherron** [reply]
> I know I have my node in the server, but how I can find it?
> 
> I've tried:
> 
>        NodeId objectId2 = NodeId.parse("ns=2;s=JoiningProcessManagement");
>    	server.getAddressSpaceManager().getManagedNode(objectId2);

You might have to cast it? If it got created as a UaObjectNode, or subclass of, it'll be in one of the AddressSpaces as one.

> how I can get addressSpace ? on the server side there is server.getAddressSpaceManager() but I think is not the same

Keep track of them as you create them.

> Yes the class ModeledAddressSpace is a subclass of ManagedAddressSpace
> but there is not method called getImplementationHandler, may be you say getInvocationHandler?

Yes, that's what I meant.

**Reply by francy328** [reply]
If I'm going to do:
 ```
NodeId objectId = NodeId.parse("ns=8;i=72");
 UaObjectNode myObject = (UaObjectNode) getNodeManager().get(objectId);
```

myObject is always null, but I know there is that node because if I'm goingo to see with UaExpert I'll find;
I do not know if I can use getNodeManager to find out a node, or if I'm going to make a mistake on NodeId.parse

**Reply by francy328** [reply]
I'm a stupid one there are two mistake:
1. The position where I put that snippet was wrong (it was before the startuptask) so when addNode to getNodeManager was't yet called;
2. and the second one is that I used get and not getNode, I solved in this way:

`NodeId objectId8 = NodeId.parse("ns=8;i=72");
    	Optional<UaNode> on = server.getAddressSpaceManager().getManagedNode(objectId8);
    	
    	UaObjectNode myObject1 = (UaObjectNode) on.get();
    	List<UaMethodNode> listMethod = myObject1.getMethodNodes();
    	 
    	listMethod.forEach((n) -> System.out.println(n.getBrowseName()));
    	
    	//trovo il metodo  
    	NodeId method = NodeId.parse("ns=8;i=97");
    	Optional<UaNode> on1 = server.getAddressSpaceManager().getManagedNode(method);
    	 UaMethodNode methodNode = (UaMethodNode) on1.get();
    	 
    	 AbortJoiningProcess aJp = new AbortJoiningProcess(methodNode);
    	 listMethod.get(0).setInvocationHandler(aJp);
    	 listMethod.get(1).setInvocationHandler(aJp);`


Now I have another problem!!!
